preview_generator/preview/builder/image__pillow.py

```diff
@@ -1,16 +1,193 @@
+        try:
+            temp_image.paste(im=origin_image, box=(0, 0), mask=origin_image)
+        except ValueError:
+            # self.logger.warning(
+            #     'Failed the transparency mask superposition. '
+            #     'Maybe your image does not contain a transparency mask')
+            temp_image.paste(origin_image)
+        temp_image.save(fp=file_output, format='jpeg', optimize=optimize, quality=quality,
+                          progressive=progressive)
+        file_output.seek(0, 0)
+        return file_output
+
+class RGBAImageConvertStrategy(TransparentImageConvertStrategy):
+    """
+    Image converter for RGBA image
+    """
+
+    def save(
+            self,
+            origin_image: PIL.Image,
+            file_output: BytesIO,
+            optimize: bool = DEFAULT_JPEG_OPTIMIZE,
+            quality: int = DEFAULT_JPEG_QUALITY,
+            progressive: bool= DEFAULT_JPEG_PROGRESSIVE
+    ) -> BytesIO:
+        return self.save_transparency(
+            origin_image=origin_image,
+            file_output=file_output,
+            optimize=optimize,
+            quality=quality,
+            progressive=progressive,
+            saving_mode='RGB',
+            background_color=DEFAULT_RGB_BACKGROUND_COLOR,
+        )
+
+class LAImageConvertStrategy(TransparentImageConvertStrategy):
+    """
+    Image converter for LA image
+    """
+    def save(
+            self,
+            origin_image: PIL.Image,
+            file_output: BytesIO,
+            optimize: bool = DEFAULT_JPEG_OPTIMIZE,
+            quality: int = DEFAULT_JPEG_QUALITY,
+            progressive: bool= DEFAULT_JPEG_PROGRESSIVE
+    ) -> BytesIO:
+        return self.save_transparency(
+            origin_image=origin_image,
+            file_output=file_output,
+            optimize=optimize,
+            quality=quality,
+            progressive=progressive,
+            saving_mode='L',
+            background_color=DEFAULT_BW_BACKGROUND_COLOR,
+        )
+
+class PillowImageConvertStrategyFactory(object):
+    """
+    Factory class to retrieve image convert strategy according to image mode
+    """
+
+    def get_strategy(self, mode: str) -> ImageConvertStrategy:
+        """
+        Get strategy to use for this image mode
+        :param mode: pillow image mode
+        :return:
+        """
+        if mode == 'RGBA':
+            return RGBAImageConvertStrategy()
+        elif mode == 'LA':
+            return LAImageConvertStrategy()
+        else:
+            return NotTransparentImageConvertStrategy()
+
 
 class ImagePreviewBuilderPillow(ImagePreviewBuilder):
+
+    def __init__(
+            self,
+            pillow_image_convert_strategy_factory: PillowImageConvertStrategyFactory = None,
+            resample_filter_algorithm: int = DEFAULT_JPEG_RESAMPLE_ALGORITHM
+    ):
+        super().__init__()
+        if pillow_image_convert_strategy_factory:
```

Comment:
better not set None as default strategy factory, doesn't it?

Reply:
actually, is there a reason for using another PillowStrategy? the factory should be instanciated at init time (I see no reason for allowing user of this class to use another factory)

Reply:
hm... you want any developer to be able to override our factory?

Reply:
You're right. There isn't anymore many reasons to change it. I did this at first because i wanted to be possible to use different save parameter, but i then change code in order to make thing more clear. I will remove this and just make it possible to use builder with parameter like optimize, progressive and quality.